src/main/java/org/rumbledb/expressions/postfix/ArrayUnboxingExpression.java

```diff
@@ -18,41 +18,39 @@
  *
  */
 
-package org.rumbledb.expressions.postfix.extensions;
+package org.rumbledb.expressions.postfix;
 
 
-import org.rumbledb.exceptions.ExceptionMetadata;
-import org.rumbledb.expressions.Expression;
-import org.rumbledb.expressions.Node;
-
 import java.util.ArrayList;
 import java.util.List;
 
+import org.rumbledb.exceptions.ExceptionMetadata;
+import org.rumbledb.expressions.Expression;
+import org.rumbledb.expressions.Node;
 
-public class PredicateExtension extends PostfixExtension {
+import sparksoniq.semantics.visitor.AbstractNodeVisitor;
 
-    private Expression expression;
+public class ArrayUnboxingExpression extends PostfixExpression {
 
-    public PredicateExtension(Expression collection, ExceptionMetadata metadata) {
-        super(metadata);
-        this.expression = collection;
+    public ArrayUnboxingExpression(Expression main, ExceptionMetadata metadata) {
+        super(main, metadata);
     }
 
-    public Expression getExpression() {
-        return this.expression;
+    @Override
+    public String serializationString(boolean prefix) {
+        String result = "(arrayUnboxing [ ])";
+        return result;
```

Comment:
I think the serialization is missing a call to the serialization of the "main" Expression

Reply:
Done, thanks a lot!